### PR TITLE
Silently ignore mono_gchandle_free(0) so callers do not have to check.

### PR DESCRIPTION
**Pull request by jaykrell**
Generally this makes for smaller easier to maintain code.

### CODE REVIEW

File: mono/metadata/threads.c [hunk 1]
```diff
@@ -2543,13 +2539,11 @@ mono_thread_internal_reset_abort (MonoInternalThread *thread)
...
 		mono_get_eh_callbacks ()->mono_clear_abort_threshold ();
 		thread->abort_exc = NULL;
-		if (thread->abort_state_handle) {
-			mono_gchandle_free (thread->abort_state_handle);
-			/* This is actually not necessary - the handle
-			   only counts if the exception is set */
-			thread->abort_state_handle = 0;
-		}
 	}
+	mono_gchandle_free (thread->abort_state_handle);
```
Comment (vargaz) [inline mono/metadata/threads.c]: This was inside an if () before, but not any more.

Comment (jaykrell) [reply mono/metadata/threads.c]: Deliberate but I can put it back.

Comment (jaykrell) [reply mono/metadata/threads.c]: oh..deliberate..and suspicious, got it.

**Comment by jaykrell** [comment]
Ping?

**Comment by jaykrell** [comment]
ping?

**Pull request merged**